cg/meta/workflow/microsalt.py

```diff
@@ -59,6 +60,19 @@ def process(self) -> Process:
             )
         return self._process
 
+    def get_case_path(self, case_id: str) -> Path:
+
+        case_obj: models.Family = self.status_db.family(case_id)
+        lims_project: str = self.get_project(case_obj)
+
+        # get the first path related to a case
+        lims_project += "_*"
+        case_path = Path(glob.glob(f"{self.root_dir}results/{lims_project}", recursive=True)[0])
```

Comment:
Should we loop over all of them and:

- check existence
- Delete if possible

I think all data from the case ids from the StatusDB analysis query should be cleaned (even old ones)

Reply:
Won't the analysis query catch all cases that can be cleaned up? For instance, the microsalt case happyhen has ben analyzed 2 times and therefore has 2 analysis objects in statusDB. This has resulted in 2 result directories:

`ACC10420_2022.9.21_18.46.0/` and `ACC10420_2022.9.11_18.45.57/`

So if the StatusDB analysis query catches both, the looping through that list in the function `past_run_dirs` should take care of it. However, I now realize that the get_case_path in microsalt should return the oldest case path and not the first index since there is a scenario where one analysis shouldn't be deleted, but the other should. I will implement this now.

Reply:
Did I address your suggestion @henrikstranneheim?

Reply:
Sounds good! Let's try it

Reply:
hmm, it seems like I was wrong regarding this. When a case is deleted, all analysis objects connected to that case will get cleaned at-date. So I will have to look into this more.

Reply:
Microsalt differs from the other pipelines by having a 1-to-1 relationship between a path and an analysis object. So every time a new analysis is run for a case, a new path will be created. 

@henrikstranneheim Is it a wanted behavior to delete all paths related to a case that is picked up by the StatusDB analysis query? So if an old analysis is picked up, even though there is a recent analysis for the same case, should we delete all paths for it or just the most recent one?

Reply:
If the most recent analysis is returned as ready for cleaning, then all paths (even old ones) should be cleaned. They are no longer interesting as it is only the newest that counts.